### PR TITLE
Microservices ebook related bug fixes.

### LINKED ISSUE #16472: IHostedService document difference
Hi,
I noticed that when I researched the working logic of method IHostedService StartAsync. Why is there a difference between two documents ?

**First document**
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=visual-studio#ihostedservice-interface

StartAsync(CancellationToken) – StartAsync contains the logic to start the background task. StartAsync is **called before**:
The app's request processing pipeline is configured (Startup.Configure).
The server is started and IApplicationLifetime.ApplicationStarted is triggered.

**Second Document** 
https://docs.microsoft.com/en-us/dotnet/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice#the-ihostedservice-interface

Specifically, start is called **after** the server has started and IApplicationLifetime.ApplicationStarted is triggered.

So I'm confused which one is right, Thank you

### LINKED ISSUE #16965: Info
@carmineelia commented on [Sun Feb 02 2020](https://github.com/dotnet/docs.it-it/issues/76)

It would seem that
the code 

services.AddDbContext&lt;CatalogContext&gt;(options =&gt;

must be replaced by

services.AddDbContext&lt;DbContext&gt;(options =&gt;

---
#### Dettagli del documento

⚠ *Non modificare questa sezione. È necessaria per i collegamenti relativi ai problemi tra docs.microsoft.com ➟ GitHub.*

* ID: 376577d9-c559-9d5d-e263-ff9f0fb6ac27
* Version Independent ID: ab3bbb06-6a80-d6d7-8060-9b93ad8b1f3e
* Content: [Creazione di un microservizio CRUD semplice basato sui dati](https://docs.microsoft.com/it-it/dotnet/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice#feedback)
* Content Source: [docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md](https://github.com/dotnet/docs.it-it/blob/live/docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md)
* Product: **dotnet-architecture**
* Technology: **microservices**
* GitHub Login: @nishanil
* Microsoft Alias: **nanil**

---

@srvbpigh commented on [Sun Feb 02 2020](https://github.com/dotnet/docs.it-it/issues/76#issuecomment-581189700)

Hello, @carmineelia
 
Thank you for your feedback.
 
We are actively reviewing your comments and will get back to you soon.

Kind regards,
Microsoft DOCS International Team

### PR DESCRIPTION
**Pull request by sughosneo**
## Summary

Describe your changes here.

Fixes #16472 - Removed repetitive defintion of IHostedService interface.
Fixes #16965 - Added right DbContext object and rephrase sentence

### CODE REVIEW

File: docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md [hunk 1]
```diff
@@ -180,14 +180,15 @@ _context.SaveChanges();
 
 In ASP.NET Core you can use Dependency Injection (DI) out of the box. You do not need to set up a third-party Inversion of Control (IoC) container, although you can plug your preferred IoC container into the ASP.NET Core infrastructure if you want. In this case, it means that you can directly inject the required EF DBContext or additional repositories through the controller constructor.
 
-In the example above of the `CatalogController` class, we are injecting an object of `CatalogContext` type plus other objects through the `CatalogController()` constructor.
+In the example above of the `CatalogController` class, we are injecting an object of `CatalogContext` which inherits from `DbContext` type and other required objects through the `CatalogController()` constructor.
```
Comment (nishanil) [inline docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md:183]:
```suggestion
In the `CatalogController` class mentioned earlier, `CatalogContext` (which inherits from `DbContext`) type is injected along with the other required objects in the `CatalogController()` constructor.
```

File: docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md [hunk 2]
```diff
@@ -180,14 +180,15 @@ _context.SaveChanges();
...
 
-An important configuration to set up in the Web API project is the DbContext class registration into the service's IoC container. You typically do so in the `Startup` class by calling the `services.AddDbContext<DbContext>()` method inside the `ConfigureServices()` method, as shown in the following **simplified** example:
+An important configuration to set up in the Web API project is the DbContext class registration into the service's IoC container. You typically do so in the `Startup` class by calling the `services.AddDbContext<CatalogContext>()` method inside the `ConfigureServices()` method, as shown in the following **simplified** example:
 
 ```csharp
 public void ConfigureServices(IServiceCollection services)
 {
     // Additional code...
+    // Actual code reference : https://github.com/dotnet-architecture/eShopOnContainers/blob/dev/src/Services/Catalog/Catalog.API/Startup.cs#L203
```
Comment (nishanil) [inline docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md:191]: remove this line

**Review by nishanil — COMMENTED** [review]
Added a few comments for you to fix.

File: docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md (hunk 2 above)
Comment (sughosneo) [reply docs/architecture/microservices/multi-container-microservice-net-applications/data-driven-crud-microservice.md]: Removed.

**Review by nishanil — APPROVED** [review]

**Pull request merged**